### PR TITLE
deprecation(semver): deprecate `SemVer` argument for `parse()` and `canParse()`

### PR DESCRIPTION
**Pull request by timreichen**
deprecates SemVer as argument type for `parse` and `canParse`.

### CODE REVIEW

File: semver/parse.ts [hunk 1]
```diff
@@ -4,6 +4,11 @@ import { isValidNumber } from "./_shared.ts";
...
 import { FULL, MAX_LENGTH, NUMERICIDENTIFIER, re, src } from "./_shared.ts";
 
+/**
+ * @deprecated (will be removed in 0.211.0) parse only parses strings
+ */
+export function parse(version: SemVer): SemVer;
+export function parse(version: string): SemVer;
 /**
  * Attempt to parse a string as a semantic version, returning either a `SemVer`
  * object or throws a TypeError.
```
Comment (kt3k) [inline semver/parse.ts:14]: I think this jsdoc block now needs to be above `function parse(version: string): SemVer;`

File: semver/format_test.ts [hunk 2]
```diff
@@ -77,7 +77,7 @@ Deno.test("format", async (t) => {
     await t.step({
       name: `format(${version} ${style} ${expected})`,
       fn: () => {
-        const v = parse(version)!;
+        const v = parse(version as SemVer)!;
```
Comment (kt3k) [inline semver/format_test.ts:80]: Why type cast to deprecated signature? I think we need to fix the test data instead

File: semver/can_parse.ts [hunk 3]
```diff
@@ -2,9 +2,13 @@
 import { SemVer } from "./types.ts";
 import { parse } from "./parse.ts";
 
+/**
+ * @deprecated (will be removed in 0.211.0) use string as an argument
+ */
+export function canParse(version: SemVer): boolean;
```
Comment (kt3k) [inline semver/can_parse.ts:8]: Looks like we also need `export function canParse(version: string): boolean;` line?

Comment (timreichen) [reply semver/can_parse.ts:8]: added

File: semver/parse.ts (hunk 1 above)
Comment (timreichen) [reply semver/parse.ts:13]: done

File: semver/can_parse.ts [hunk 4]
```diff
@@ -2,9 +2,14 @@
 import { SemVer } from "./types.ts";
 import { parse } from "./parse.ts";
 
+/**
+ * @deprecated (will be removed in 0.211.0) use string as an argument
```
Comment (iuioiua) [inline semver/can_parse.ts:6]:
```suggestion
 * @deprecated (will be removed in 0.211.0) Use a string argument instead.
```

File: semver/parse.ts [hunk 5]
```diff
@@ -4,12 +4,17 @@ import { isValidNumber } from "./_shared.ts";
 import { isSemVer } from "./is_semver.ts";
 import { FULL, MAX_LENGTH, NUMERICIDENTIFIER, re, src } from "./_shared.ts";
 
+/**
+ * @deprecated (will be removed in 0.211.0) parse only parses strings
```
Comment (iuioiua) [inline semver/parse.ts:8]:
```suggestion
 * @deprecated (will be removed in 0.211.0) Use a string argument instead.
```

**Review by iuioiua — CHANGES_REQUESTED** [review]

**Review by iuioiua — APPROVED** [review]
LGTM. Nice! Thanks, Tim.

File: semver/can_parse.ts [hunk 6]
```diff
@@ -2,9 +2,14 @@
 import { SemVer } from "./types.ts";
 import { parse } from "./parse.ts";
 
+/**
+ * @deprecated (will be removed in 0.211.0) Use a string argument instead.
```
Comment (iuioiua) [inline semver/can_parse.ts:6]:
```suggestion
 * @deprecated (will be removed in 0.212.0) Use a string argument instead.
```

File: semver/parse.ts [hunk 7]
```diff
@@ -4,12 +4,17 @@ import { isValidNumber } from "./_shared.ts";
 import { isSemVer } from "./is_semver.ts";
 import { FULL_REGEXP, MAX_LENGTH } from "./_shared.ts";
 
+/**
+ * @deprecated (will be removed in 0.211.0) Use a string argument instead.
```
Comment (iuioiua) [inline semver/parse.ts:8]:
```suggestion
 * @deprecated (will be removed in 0.212.0) Use a string argument instead.
```

**Review by iuioiua — CHANGES_REQUESTED** [review]

**Review by iuioiua — APPROVED** [review]

**Pull request merged**